pay-api/src/pay_api/models/base_model.py

```diff
@@ -85,5 +85,7 @@ def find_by_id(cls, identifier: int) -> Self:
     def find_by_id_for_update(cls, identifier: int) -> Self:
         """Return model by id with lock to avoid race conditions."""
         if identifier:
-            return cls.query.filter_by(id=identifier).with_for_update().one_or_none()
+            query = cls.query.filter_by(id=identifier).with_for_update()
+            query = query.populate_existing()
```

Comment:
If you have an existing fetch of eg `payment_account.find_by_id`, this will refresh the row, if you don't do this you may have old data which causes a row conflict, we didn't see this in the versioned unit test because it only grabbed the data once